xarray/backends/api.py

```diff
@@ -549,6 +549,25 @@ def to_netcdf(dataset, path=None, mode='w', format=None, group=None,
             engine = _get_default_engine(path)
         path = _normalize_path(path)
 
+    for var in encoding:
```

Comment:
Can you put this logic in `conventions.py` instead, [after this line](https://github.com/pydata/xarray/blob/7084df0e82c3c78765691910cd02b197b5772575/xarray/conventions.py#L726) where we converted to any encoded dtype? If `_FillValue` was set in `encoding` at the point, it has already been converted into `atttrs`.